src/main/java/com/marek/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -12,12 +10,7 @@
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+
     @Query("SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :categoriesId")
     List<Book> findAllByCategoriesId(Long categoriesId, Pageable pageable);
```

Comment:
The method findAllByCategoriesId returns a List, but the method signature suggests it should return a Page. You might want to adjust either the method signature or the return statement to match the type.